monitor-manager/monitor-manager.py

```diff
@@ -0,0 +1,76 @@
+from csv import reader
+from flask import Flask
+import socket
+import json
+import requests
+import time
+
+app = Flask(__name__)
+
+
+@app.route('/', methods=['get'])
+def health_check():
+    return "Monitor Manager working"
+
+
+@app.route('/experiments', methods=['post'])
+def start_experiments():
+    experiments_plan = read_csv("experiments-plan.csv")
+    hosts_info = read_csv("hosts-info.csv")
+    is_spark_node, host_ip = check_if_spark_node(hosts_info)
+
+    for function_name, table in experiments_plan:
+        print("Task submitting")
+        submit_response = requests.post("http://192.168.55.20:5000/submit", json={"function_name": function_name})
+        submit_response = json.loads(submit_response.text)
+        driver_id = submit_response["submissionId"]
+        print("Submit driver ID: ", driver_id)
+
+        print("Start monitoring")
+        system_monitor(host_ip, hosts_info, function_name, True)
+        #
+        status = "RUNNING"
+        while status == "RUNNING":
+            status_response = requests.get("http://192.168.55.20:5000/status", json={"driver_id": driver_id})
+            status_response = json.loads(status_response.text)
+            status = status_response["driverState"]
+            time.sleep(1)
+
+        system_monitor(host_ip, hosts_info, function_name, False)
+
+    return "Monitoring finished", 200
+
+
+def system_monitor(host_ip, hosts_info, function_name, start):
+    for ip, container_name in hosts_info:
+        url = "http://" + str(ip) + ":8063/monitor"
+        request_data = {
+            "container_name": container_name,
+            "function_name": function_name
+        }
+        if start:
+            response = requests.post(url, json=request_data)
+            print(response.text)
+        else:
+            response = requests.delete(url)
+        print("IP: ", ip, " ", response.text)
+
+
+def read_csv(filename):

```

Comment:
why not pandas csv reader? It's not like I would like it to be changed, but I'm just asking a question - I might be missing something

Reply:
no reason, i can change it to pandas